src/pieces/commands/run_assets_command.py

```diff
@@ -0,0 +1,78 @@
+from pieces.settings import Settings
+from pieces.assets import check_assets_existence, AssetsCommandsApi
+from typing import List, Tuple, Callable
+import subprocess
+from enum import Enum
+from list_command import PiecesSelectMenu
+from pieces.assets.assets_command import AssetsCommands
+
+class AssetClassification(Enum):
+    SHELL = "sh"
+    BASH = "bat"  # Added BASH classification
+
+class ExecuteCommand:
+    @classmethod
+    @check_assets_existence
+    def execute_command(cls, max_assets: int = 10):
+        if max_assets < 1:
+            print("Max assets must be greater than 0")
+            max_assets = 10
+        assets_snapshot = AssetsCommandsApi().assets_snapshot
+        assets = []
+        
+        for i, uuid in enumerate(list(assets_snapshot.keys())[:max_assets], start=1):
+            asset = AssetsCommandsApi.get_asset_snapshot(uuid)
+            classification = cls.get_asset_classification(asset)
+            if classification in [AssetClassification.SHELL.value, AssetClassification.BASH.value]:
+                assets.append((f"{i}: {asset.name}", {"ITEM_INDEX": i, "UUID": uuid, "CLASSIFICATION": classification}))
+        
+        if not assets:
+            print("No shell or bash assets found")
+            return
+        
+        def custom_callback(**kwargs):
```

Comment:
Can we rename it to something better than that?

Reply:
addressed issue in commit [c0d8d13](https://github.com/pieces-app/cli-agent/pull/169/commits/c0d8d139784b9a1296dbbb0771bb40013df3f785) . renamed custom_callback to open_and_execute_asset .  this change aims to make the function more self explanatory .